src/test/java/com/exasol/adapter/dialects/SqlGenerationVisitorTest.java

```diff
@@ -225,4 +226,126 @@ void testVisitSqlLiteralDate() {
         final SqlLiteralDate sqlLiteralDate = new SqlLiteralDate("2015-12-01");
         assertThat(sqlGenerationVisitor.visit(sqlLiteralDate), equalTo("DATE '2015-12-01'"));
     }
+
+    @Test
+    void testVisitSqlFunctionAggregateListaggQuoting() throws AdapterException {
+        final List<SqlNode> arguments = new ArrayList<>();
+        arguments.add(new SqlColumn(1, ColumnMetadata.builder().name("a \"'").type(DataType.createBool()).build()));
+        final SqlFunctionAggregateListagg.Behavior overflowBehavior = new SqlFunctionAggregateListagg.Behavior(
+                SqlFunctionAggregateListagg.BehaviorType.TRUNCATE);
+        overflowBehavior.setTruncationType("WITH COUNT");
+        overflowBehavior.setTruncationFiller("\" filler '");
+        final List<SqlNode> expressions = new ArrayList<>();
+        expressions.add(new SqlColumn(1, ColumnMetadata.builder().name("b \"'").type(DataType.createBool()).build()));
+        final SqlOrderBy orderBy = new SqlOrderBy(expressions, List.of(false), List.of(true));
+        final SqlFunctionAggregateListagg listagg = SqlFunctionAggregateListagg.builder(arguments, overflowBehavior)
+                .orderBy(orderBy).separator(", ").build();
```

Comment:
Please use something like `"\" separator '"`.

Reply:
Do you mean placing the quoting marks far from each other?

Reply:
I tried to highlight only the last line and meant this:
```suggestion
                .orderBy(orderBy).separator("\" separator '").build();
```